include/boost/url/authority_view.hpp

```diff
@@ -529,26 +530,13 @@ class BOOST_SYMBOL_VISIBLE
             @ref has_userinfo,
             @ref encoded_userinfo.
     */
-    template<
-        class Allocator =
-            std::allocator<char>>
-    const_string
-    userinfo(
-        Allocator const& a = {}) const
+    pct_encoded_view
+    userinfo() const
     {
         pct_decode_opts opt;
         opt.plus_to_space = false;
-        string_view s = encoded_userinfo();
-        std::size_t n =
-            pct_decode_bytes_unchecked(s);
-        return const_string(
-            n, a,
-            [&s, &opt](
-                std::size_t n, char* dest)
-            {
-                pct_decode_unchecked(
-                    dest, dest + n, s, opt);
-            });
+        return pct_encoded_view(
+            encoded_userinfo(), opt);
```

Comment:
We should open an issue and note that this function could use the `decoded_` array information